File: upl-purchase-sdk-php/src/Api/LegalDocumentsApi.php
<?php

/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */

declare(strict_types=1);

namespace Unzer\PayLater\Api;

use Unzer\PayLater\Communication\HttpMethod;
use Unzer\PayLater\Communication\RequestHeader;
use Unzer\PayLater\Communication\RequestHeaderCollection;
use Unzer\PayLater\Exception\ApiResponseException;
use Unzer\PayLater\Exception\BuilderException;
use Unzer\PayLater\Exception\ResponseException;

class LegalDocumentsApi extends BaseApi
{
    /**
     * Generates a terms-and-conditions document in html format.
     *
     * @param string $purchaseId The purchaseId received from the initialize request that started the verification process.
     * @param string $authorization The access token received from the initialize request. Provide this for client-side requests in the Bearer format.
     * @return string Terms and conditions in HTML format.
     * @throws ApiResponseException
     * @throws BuilderException
     */
    public function getTermsAndConditionsWithAuthorization(
        string $purchaseId,
        string $authorization
    ): string {
        $uri = '/purchase/legaldocuments/termsandconditions/{purchaseId}';
        $uri = $this->populateUri($uri, 'purchaseId', $purchaseId);

        $requestHeaderCollection = new RequestHeaderCollection(
            new RequestHeader('Authorization', 'Bearer ' . $authorization)
        );

        try {
            return $this->communicator->getStringResponse(
                HttpMethod::GET,
                $uri,
                $requestHeaderCollection,
                null
            );
        } catch (ResponseException $exception) {
            throw $this->createApiResponseException($exception);
        }
    }

    /**
     * Generates a terms-and-conditions document in html format.
     *
     * @param string $purchaseId The purchaseId received from the initialize request that started the verification process.
     * @param string $unzerPlSecretKey Secret key which can be requested from your account manager. Provide this for server-to-server communication.
     * @return string Terms and conditions in HTML format.
     * @throws ApiResponseException
     * @throws BuilderException
     */
    public function getTermsAndConditions(
        string $purchaseId,
        string $unzerPlSecretKey
    ): string {
        $uri = '/purchase/legaldocuments/termsandconditions/{purchaseId}';
        $uri = $this->populateUri($uri, 'purchaseId', $purchaseId);

        $requestHeaderCollection = new RequestHeaderCollection(
            new RequestHeader('unzer-pl-secret-key', $unzerPlSecretKey)
        );

        try {
            return $this->communicator->getStringResponse(
                HttpMethod::GET,
                $uri,
                $requestHeaderCollection,
                null
            );
        } catch (ResponseException $exception) {
            throw $this->createApiResponseException($exception);
        }
    }
}

File: upl-purchase-sdk-java/src/main/java/com/unzer/paylater/api/LegalDocumentsApi.java
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */
package com.unzer.paylater.api;

import java.util.ArrayList;
import java.util.List;

import com.unzer.paylater.communication.Communicator;
import com.unzer.paylater.communication.HttpMethod;
import com.unzer.paylater.communication.RequestHeader;
import com.unzer.paylater.exception.ResponseException;

public class LegalDocumentsApi extends BaseApi {

    public LegalDocumentsApi(Communicator communicator) {
        super(communicator);
    }

    /**
     * Generates a terms-and-conditions document in html format.
     *
     * @param purchaseId The purchaseId received from the initialize request that started the verification process.
     * @param authorization The access token received from the initialize request. Provide this for client-side requests in the Bearer format.
     * @return String - Terms and conditions in HTML format.
     */
    public String getTermsAndConditionsWithAuthorization(String purchaseId, String authorization) {
        String uri = "/purchase/legaldocuments/termsandconditions/{purchaseId}";
        uri = populateUri(uri, "purchaseId", purchaseId);

        List<RequestHeader> headerParams = new ArrayList<>();
        addHeaderParam(headerParams, "Authorization", authorization);

        try {
            return communicator.execute(
                    HttpMethod.GET,
                    uri,
                    headerParams,
                    null,
                    String.class);
        } catch (ResponseException e) {
            throw createException(e);
        }
    }

    /**
     * Generates a terms-and-conditions document in html format.
     *
     * @param purchaseId The purchaseId received from the initialize request that started the verification process.
     * @param unzerPlSecretKey Secret key which can be requested from your account manager. Provide this for server-to-server communication.
     * @return String - Terms and conditions in HTML format.
     */
    public String getTermsAndConditions(String purchaseId, String unzerPlSecretKey) {
        String uri = "/purchase/legaldocuments/termsandconditions/{purchaseId}";
        uri = populateUri(uri, "purchaseId", purchaseId);

        List<RequestHeader> headerParams = new ArrayList<>();
        addHeaderParam(headerParams, "unzer-pl-secret-key", unzerPlSecretKey);

        try {
            return communicator.execute(
                    HttpMethod.GET,
                    uri,
                    headerParams,
                    null,
                    String.class);
        } catch (ResponseException e) {
            throw createException(e);
        }
    }
}
